src/server/server_family.cc

```diff
@@ -2258,6 +2258,7 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
   }
 
   if (should_enter("REPLICATION")) {
+    unique_lock lk(replicaof_mu_);
```

Comment:
if you take `replicaof_mu_`, you should not rely here on `etl.is_master` and you can check `replica_` instead.
also, a comment saying that `etl.is_master` is eventually consistent and does not fit here would explain why you access `replica_` directly.